dotnet/src/Connectors/Connectors.MistralAI/Client/MistralClient.cs

```diff
@@ -33,52 +34,89 @@ internal sealed class MistralClient
 
-        return response.Choices.Select(chatChoice => new ChatMessageContent(new AuthorRole(chatChoice.Message!.Role), chatChoice.Message!.Content, this._modelId, chatChoice, Encoding.UTF8, GetChatChoiceMetadata(response, chatChoice))).ToList();
+        return response.Choices.Select(chatChoice => new ChatMessageContent(new AuthorRole(chatChoice.Message!.Role!), chatChoice.Message!.Content, this._modelId, chatChoice, Encoding.UTF8, GetChatChoiceMetadata(response, chatChoice))).ToList();
     }
 
-    internal async IAsyncEnumerable<StreamingChatMessageContent> GetStreamingChatMessageContentsAsync(ChatHistory chatHistory, CancellationToken cancellationToken, PromptExecutionSettings? executionSettings = null, Kernel? kernel = null)
+    internal async IAsyncEnumerable<StreamingChatMessageContent> GetStreamingChatMessageContentsAsync(ChatHistory chatHistory, [EnumeratorCancellation] CancellationToken cancellationToken, PromptExecutionSettings? executionSettings = null, Kernel? kernel = null)
     {
         string modelId = executionSettings?.ModelId ?? this._modelId;
         var mistralExecutionSettings = MistralAIPromptExecutionSettings.FromExecutionSettings(executionSettings);
-        var request = this.CreateChatCompletionRequest(modelId, false, chatHistory, mistralExecutionSettings);
-        var endpoint = new Uri($"{this._endpoint}{this._separator}chat/completions");
+        var request = this.CreateChatCompletionRequest(modelId, stream: true, chatHistory, mistralExecutionSettings);
+        var endpoint = this.GetEndpoint(mistralExecutionSettings, path: "chat/completions");
 
         using var httpRequestMessage = this.CreatePost(request, endpoint, this._apiKey, stream: true);
 
         using var response = await this.SendStreamingRequestAsync(httpRequestMessage, cancellationToken).ConfigureAwait(false);
 
         using var responseStream = await response.Content.ReadAsStreamAndTranslateExceptionAsync().ConfigureAwait(false);
-        foreach (var streamingTextContent in this.ProcessResponseStream(responseStream, modelId))
+        using var reader = new StreamReader(responseStream);
+        string line;
+        string? rawChunk = null;
+        AuthorRole? currentRole = null;
+        while ((line = await reader.ReadLineAsync().ConfigureAwait(false)) != null)
         {
-            yield return streamingTextContent;
+            if (!string.IsNullOrEmpty(line))
+            {
+                rawChunk = line.Substring("data:".Length).Trim();
+            }
+            else
+            {
+                if (rawChunk is not null and "[DONE]")
+                {
+                    continue;
+                }
+                var chunk = await JsonSerializer.DeserializeAsync<MistralChatCompletionChunk>(
+                    new MemoryStream(Encoding.UTF8.GetBytes(rawChunk))).ConfigureAwait(false);
```

Comment:
Any special reason why are you converting the string to a stream? 
AFAIK `DeserializeAsync<T>` accepts string as input.

```suggestion
                var chunk = await JsonSerializer.DeserializeAsync<MistralChatCompletionChunk>(rawChunk).ConfigureAwait(false);
```

Reply:
It only takes a stream

Reply:
Ah ok, it seems an overkill creating an stream just to use the `DeserializeAsync`, can't we just use the synchronous one and avoid allocating a stream?